src/models.py

```diff
@@ -15,6 +21,7 @@ class User(Base):
     id = Column(Integer, primary_key=True)  # noqa: A003
     username = Column(String, nullable=False, unique=True)
     password_hash = Column(String(64), nullable=False)
+    topics: List[Topic] = relationship("Topic", backref="author", lazy="dynamic")
```

Comment:
Seems like it's better to add such relationship to `Topic` model. Since we want to each topic have a user associated with it. Like this:
```python
class Topic(Base):
    id = Column(Integer, primary_key=True)  # noqa: A003
    author_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    ...
    author: User = relationship("User", uselist=False)
```
Because seems like that is exactly what we want, and allow us to explicitly express our intentions.

And yes, your approach works totally fine, but seems just a little bit implicit in terms of our intentions.

Reply:
SQLAlchemy documentation says:
`A one to many relationship places a foreign key on the child table referencing the parent. **relationship() is then specified on the parent**, as referencing a collection of items represented by the child`
And:
`...relationship() at mapper configuration time, based on the type and direction of the **relationship - one to many forms a list**, many to one forms a scalar, many to many is a list.`

That seems like a complicated topic. So far I've been treating documentation of any kind as a Holy scripture, and here we commit cardinal sin. It isn't that I question your expertise, it's that the difference between documentation and this solution is so drastic it's just puzzling. Should we discuss it in more detail, or should I just follow your instructions and it will become clear later on?

Reply:
Yes, relationships can be tricky and confusing sometimes. I think we can proceed with my solution for now, and after that have a discussion to get answers for questions.

Reply:
> difference between documentation and this solution is so drastic

I see no difference between docs and this solution. What differences do you mean?